spyder/plugins/projects/widgets/main_widget.py

```diff
@@ -1015,55 +995,111 @@ def _get_valid_recent_projects(self, recent_projects):
 
         return valid_projects
 
-    def _execute_fzf_subprocess(self, search_text=""):
+    # ---- Private API for the Switcher
+    # -------------------------------------------------------------------------
+    def _call_fzf(self, search_text=""):
         """
-        Execute fzf subprocess to get the list of files in the current
-        project filtered by `search_text`.
+        Call fzf in a worker to get the list of files in the current project
+        that match with `search_text`.
 
         Parameters
         ----------
-        search_text: str
-            The current search text in the switcher dialog box.
+        search_text: str, optional
+            The search text to pass to fzf.
         """
         project_path = self.get_active_project_path()
-        if project_path is None:
-            return []
+        if self._fzf is None or project_path is None:
+            return
+
+        self._worker_manager.terminate_all()
+
+        worker = self._worker_manager.create_process_worker(
+            [self._fzf, "--filter", search_text],
+            os.environ.copy()
+        )
+
+        worker.set_cwd(project_path)
+        worker.sig_finished.connect(self._process_fzf_output)
+        worker.start()
+
+    def _process_fzf_output(self, worker, output, error):
+        """Process output that comes from the fzf worker."""
+        if output is None or error:
+            return
+
+        relative_path_list = output.decode('utf-8').strip().split("\n")
+        project_path = self.get_active_project_path()
+
+        # List of tuples with the absolute path
+        result_list = [
+            osp.normpath(os.path.join(project_path, path))
+            for path in relative_path_list
+        ]
```

Comment:
Not sure if should be done at this point in the code, but I think the project path itself needs to be skipped. Checking locally, I'm able to see an entry for the root directory of the project and clicking in it causes to open the directory in the system files explorer :

![switcher_dir](https://github.com/spyder-ide/spyder/assets/16781833/bd329550-c915-4838-9ed1-ff793e61ac95)

Or maybe that's the intended behavior?

Reply:
Great catch @dalthviz! This problem happened only on Windows due to the use of `os.normpath` on an empty string. And I fix it in one of my last commits.